src/software/ai/hl/stp/play/ball_placement/ball_placement_play_fsm.cpp

```diff
@@ -220,55 +329,57 @@ bool BallPlacementPlayFSM::retreatDone(const Update &event)
                                        ->position()) > RETREAT_DISTANCE_METERS;
 }
 
-Angle BallPlacementPlayFSM::calculateWallKickoffAngle(const Point &ball_pos,
-                                                      const Rectangle &field_lines)
+std::pair<Angle, Point> BallPlacementPlayFSM::calculateWallPickOffLocation(const Point &ball_pos,
+                                                      const Rectangle &field_boundary,
+                                                      const double max_dist)
 {
-    Angle kick_angle;
-    if (ball_pos.x() > field_lines.xMax())
+    Angle facing_angle;
+    Point backoff_point;
+    bool near_north = field_boundary.yMax() - ball_pos.y() < max_dist;
+    bool near_south = ball_pos.y() - field_boundary.yMin() < max_dist;
+    bool near_east = field_boundary.xMax() - ball_pos.x() < max_dist;
+    bool near_west = ball_pos.x() - field_boundary.xMin() < max_dist;
```

Comment:
north, south, east, west isn't part of our conventions. I'd suggest naming it as `near_*_goal_line` and `near_{dir}_y_boundary`